main.go

```diff
@@ -74,5 +82,14 @@ func main() {
 	chkfatal(err)
 	srv := makeHandler(&config, NewDaemon(state))
 	http.Handle("/", srv)
-	chkfatal(http.ListenAndServe(config.ListenAddr, nil))
+	if config.WebProtocol == "http" {
+		chkfatal(http.ListenAndServe(config.ListenAddr, nil))
+	} else if config.WebProtocol == "https" {
+		chkfatal(http.ListenAndServeTLS(config.ListenAddr,
+			config.TLSCert,
+			config.TLSKey,
+			nil))
+	} else {
+		chkfatal(ErrUnknownProtocol)
```

Comment:
I think definining an error constant up top and then using it here is an excessive layer of indirection; it's not like we're using the information programmatically or reporting this error anywhere else, and it just makes things harder to follow. chkfatal is also kindof silly here, as you've already done the check. Why not just `log.Fatal("Error: WebProtocol must be either http or https.")`?

Reply:
Sure, that works

Reply:
Yeah, switching to a boolean obviates this; probably fine to just trust the marshalling errors there.